R/ipc_get_population.R

```diff
@@ -83,278 +84,28 @@ ipc_get_population <- function(
-    renamed_groups_df <- renamed_df %>%
-      dplyr::filter(
-        !sapply(.data$groups, is.null)
-      ) %>%
-      dplyr::select(
-        -c(dplyr::any_of("areas"), dplyr::matches("^phase|^estimated"))
-      ) %>%
-      dplyr::mutate(
-        "groups" := purrr::map(.data$groups, dplyr::as_tibble)
-      ) %>%
-      tidyr::unnest(
-        cols = "groups"
-      ) %>%
-      rename_population_df() %>%
-      dplyr::rename(
-        "group_id" := "id",
-        "group_name" := "name"
-      )
-
-    groups_df <- renamed_groups_df %>%
-      dplyr::select(-dplyr::any_of("areas")) %>%
-      dplyr::distinct() %>%
-      pivot_population_df() %>%
-      dplyr::filter(.data$period_dates != "") %>%
-      create_date_columns() %>%
-      arrange_population_df()
-  } else {
-    groups_df <- NULL
-  }
-
-  # extract areas data frame
-  # have to do in two stages, extracting areas that don't have groups
-  # then extracting again from areas under groups
-  if ("areas" %in% names(renamed_df) && !all(is.na(renamed_df$areas))) {
-    renamed_areas_df1 <- renamed_df %>%
-      dplyr::filter(
-        !sapply(.data$areas, is.null)
-      ) %>%
-      dplyr::select(
-        -c(dplyr::any_of("groups"), dplyr::matches("^phase|^estimated"))
-      ) %>%
-      dplyr::mutate(
-        "areas" := purrr::map(.data$areas, dplyr::as_tibble)
-      ) %>%
-      tidyr::unnest(
-        cols = "areas"
-      ) %>%
-      rename_population_df()
-  } else {
-    renamed_areas_df1 <- NULL
-  }
-
-
-  # now extract areas that are under groups
-  if ("groups" %in% names(renamed_df) && "areas" %in% names(renamed_df) && !all(is.na(renamed_df$areas))) {
-    renamed_areas_df2 <- renamed_groups_df %>%
-      dplyr::filter(
-        !sapply(.data$areas, is.null)
-      ) %>%
-      dplyr::mutate(
-        "areas" := purrr::map(.data$areas, dplyr::as_tibble)
-      ) %>%
-      dplyr::select(
-        -c(dplyr::matches("^phase|^estimated"))
-      ) %>%
-      tidyr::unnest(
-        cols = "areas"
-      ) %>%
-      rename_population_df()
-  } else {
-    renamed_areas_df2 <- NULL
-  }
-
-
-  # now combine into a single areas dataset
-  # TODO: remove `-dplyr::starts_with("group")` once
-  # the IPC team fixes the API, currently it duplicates
-  # all areas in a country for each group
-  if (!all(c(is.null(renamed_areas_df1), is.null(renamed_areas_df2)))) {
-    areas_df <- dplyr::bind_rows(renamed_areas_df1, renamed_areas_df2) %>%
-      pivot_population_df() %>%
-      dplyr::filter(
-        .data$period_dates != ""
-      ) %>%
-      create_date_columns() %>%
-      dplyr::distinct(
-        dplyr::across(
-          -dplyr::starts_with("group") # TODO: remove once API is fixed
-        )
-      ) %>%
-      dplyr::rename(
-        "area_id" := "id",
-        "area_name" := "name"
-      ) %>%
-      arrange_population_df()
-  } else {
-    areas_df <- NULL
+  if (!all(sapply(ret, inherits, what = "list"))) {
```

Comment:
Just to avoid using `purrr:::()`. I like `sapply`, it's usually the other functions that I enjoy using. Anyway, your idea for functionalizing is good.